### ISSUE TITLE
Doesn't work with styled components

### ISSUE THREAD

**Issue by fastfedora** [stylelint/postcss-css-in-js#76]
Not sure which project to add an issue to, since I've been bouncing around for the past 2 hours trying to figure out how `stylelint` interacts with styled components and trying to untangle how to get setup properly.

I inherited a codebase that I'm trying to add `stylelint` into. However, it only reports on lint errors if I'm using either:

- `stylelint-processor-styled-components` as the processor
- `postcss-styled` passed in as a custom syntax

If I try to use `stylelint` with no processor or custom syntax, or use `--syntax css-in-js` to force this syntax to be used, no lint errors are reported and `stylelint` completes with no errors.

There's several comments on various issues indicating that `stylelint-processor-styled-components` is deprecated, or should be deprecated, and it appears that `postcss-styled` was removed as the default in `stylelint` in favor of `postcss-jsx` (now `postcss-css-in-js`). 

But, apparently, that doesn't work. 

For example, when running `stylelint --customSyntax postcss-styled './src/**/*.jsx'`, with the following `.stylelintrc.json`:

```
{
	"extends": [
		"stylelint-config-recommended",
		"stylelint-config-styled-components"
	]
}
```

And the following `Example.jsx`:

```
const Example = styled.div`
	margin-bottom: 0;
	margin: 25px 0 35px 0;
`;
```

The following is returned:

```
src/components/Example.jsx
 6:2  ✖  Unexpected shorthand "margin" after "margin-bottom"   declaration-block-no-shorthand-property-overrides
```

Running the same configuration, with either an explicit `--syntax` or no options at all, results in no output:

```
> stylelint './src/**/*.jsx'
> stylelint --syntax css-in-js './src/**/*.jsx'    
```

If I add `stylelint-processor-styled-components` and change my configuration to:

```
{
	"processors": [
		"stylelint-processor-styled-components"
	],
	"extends": [
		"stylelint-config-recommended",
		"stylelint-config-styled-components"
	]
}
```

Then I once again get the lint error. 

It would be nice to use `stylelint` out-of-the-box, as has been suggested, but it does not appear to be working. I would normally just use one of the other solutions, but I'm trying to get double-slash comments to work, and neither `stylelint-processor-styled-components` nor `postcss-styled` appears to support them for now.

Sorry for the vague bug report. I've now gone through all of the above repos, looked at issues, and tried a bunch of different combinations to make things work. I can only assume there is some bug that is preventing `postcss-css-in-js` from working with styled components. Hopefully the above example can help track down where.

**Comment by bingtsingw** [comment]
I have the same issue.

**Comment by GollyJer** [comment]
Does anyone have styled components working?
Can't figure out if I'm doing something wrong or if it just doesn't work.

**Comment by mattdarveniza** [comment]
Yeah there is next to zero documentation for how you are actually supposed to get this to work with styled components using a `.stylelintrc`. The examples in the readme aren't helpful because they're talking about configuring styled components at the actual point where the linter is called in node as far as I can tell.

Some documentation on how to configure this plugin to work in with a RC file would be very helpful.

**Comment by adbutterfield** [comment]
Another thing to note, I had to additionally install `postcss-syntax` as well to get this to work.

**Comment by tobilen** [comment]
This does not work for me, which may be related to me using yarn 3.

Input:
```
module.exports = {
  customSyntax: "@stylelint/postcss-css-in-js",
  extends: ["stylelint-config-standard"],
};
```

Output:
```
$: yarn stylelint
Error: Cannot resolve custom syntax module "@stylelint/postcss-css-in-js". Check that module "@stylelint/postcss-css-in-js" is available and spelled correctly.
    at getCustomSyntax (/Users/tobilen/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/getPostcssResult.js:99:11)
    at getPostcssResult (/Users/tobilen/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/getPostcssResult.js:41:5)
    at lintSource (/Users/tobilen/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/lintSource.js:76:20)
    at async /Users/tobilen/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/standalone.js:218:27
    at async Promise.all (index 0)
    at async standalone (/Users/tobilen/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/standalone.js:257:22)
```

Doesn't work when resolving the path ourselves with yarns resolve (https://yarnpkg.com/features/pnp#packages-are-stored-inside-zip-archives-how-can-i-access-their-files) either:
Input:
```
module.exports = {
  customSyntax: require.resolve("@stylelint/postcss-css-in-js"),
  extends: ["stylelint-config-standard"],
};
```

Output:
```
$: yarn stylelint
Error: Cannot resolve custom syntax module "/Users/tobilen/projects/project/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-8be34fbed8/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/index.js". Check that module "/Users/tobilen/projects/project/.yarn/__virtual__/@stylelint-postcss-css-in-js-virtual-8be34fbed8/0/cache/@stylelint-postcss-css-in-js-npm-0.37.2-8a19c1b9be-cc9b5d1bd9.zip/node_modules/@stylelint/postcss-css-in-js/index.js" is available and spelled correctly.
    at getCustomSyntax (/Users/tobilen/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/getPostcssResult.js:99:11)
    at getPostcssResult (/Users/tobilen/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/getPostcssResult.js:41:5)
    at lintSource (/Users/tlengsholz/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/lintSource.js:76:20)
    at async /Users/tlengsholz/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/standalone.js:218:27
    at async Promise.all (index 0)
    at async standalone (/Users/tlengsholz/projects/project/.yarn/cache/stylelint-npm-14.0.1-9690b0ed70-7279f25282.zip/node_modules/stylelint/lib/standalone.js:257:22)
```

maybe related to https://github.com/stylelint/stylelint/issues/5632?

**Comment by tobilen** [comment]
> @tobilen out of curiosity, are you using parcel@v2? My above solution stopped working and your mentioned error appeared once we installed parcel **anywhere** in our monorepo project… 🤔

nope, no parcel anywhere in the repo.

**Comment by melink14** [comment]
Interestingly, this worked when I used npm v8 but when I downgraded to v6 for other reasons I started getting this exact error. (No other changes in the branch other than to use customSyntax and upgrade to stylelint 14)

**Comment by brandondurham** [comment]
Experiencing the same issue as @tobilen. Tried yarn versions 1.22, 2.x and 3.x.
